code/API_definitions/location-subscriptions.yaml

```diff
@@ -0,0 +1,465 @@
+        - AREA_LEFT
+    
+    NotificationUrl:
+      type: string
+      example: "https://application-server.com"
+      description: https callback address where the event notification must be POST-ed
+    
+    NotificationAuthToken:
+      type: string
+      example: "c8974e592c2fa383d4a3960714"
+      description: Authentication token for callback API 
+    
+    CreateEventSubscription:
+      description: The request for creating a Location Event Subscription
+      type: object
+      properties:
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+      required:
+        - webhook
+        
+    EventSubscriptionInfo:
+      description: Represents a area-location event subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateEventSubscription'
+        - type: object
+          properties:
+            eventSubscriptionId:
+              $ref: '#/components/schemas/EventSubscriptionId'
+            webhook:
+              $ref: '#/components/schemas/Webhook'
+            subscriptionDetail:
+              $ref: '#/components/schemas/SubscriptionDetail'
+            subscriptionExpireTime:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+            startsAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be started (e.g. in 7 days from now on). Provided in date-time format.
+            expiresAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be expired (e.g. in 7 days from now on). Provided in date-time format.
+          required:
+            - eventSubscriptionId
+            - webhook
+  
+    EventTime:
+      format: date-time
+      type: string
+      description: The time when the event was notified.
+    
+    EventNotifications:
+      description: The event notification callback.
+      type: object
+      required:
+        - event
+      properties:
+        eventSubscriptionId:
+          $ref: '#/components/schemas/EventSubscriptionId'
+        event:
+          $ref: '#/components/schemas/Event'
+    
+    Event:
+      description: The event being notified
+      anyOf:
+        - $ref: '#/components/schemas/FindLocationDeviceEvent'
```

Comment:
What does FindLocationDeviceEvent mean? We have 2 events area entered and area left. So we can make 2 events AreaEnteredEvent, AreaLeftEvent or just one like AreaEvent, but I tend to 2 events.

Reply:
This is just a sample which return point(longlitude, latitude) of device. But I can instead of it make two another event as we mentioned.